src/frontend/src/api/backend-sdk/sendToken.ts

```diff
@@ -0,0 +1,107 @@
+import { auth } from '@clerk/nextjs/server';
+
+const CHAT_SERVICE_HOST = process.env.CHAT_SERVICE_HOST;
+const CHAT_SERVICE_PORT = process.env.CHAT_SERVICE_PORT;
+
+/**
+ * Global cache for storing orgId and GUID mappings (ensures unique values)
+ */
+const customerCache = new Map<string, string>();
+
+/**
+ * Centralized logger function
+ */
+function logMessage(level: 'info' | 'warn' | 'error', message: string, ...args: any[]) {
+  if (process.env.NODE_ENV !== 'production' || level === 'error') {
+    // eslint-disable-next-line no-console
+    console[level](`[${level.toUpperCase()}] ${message}`, ...args);
+  }
+}
+
+/**
+ * Log the current state of customerCache
+ */
+function logCustomerCache() {
+  logMessage('info', 'Current customerCache contents:', Array.from(customerCache.entries()));
+}
+
+/**
+ * Fetch customer GUID for the authenticated user (Ensures API is called only once per session)
+ */
+export async function getCustomerGuid(): Promise<string | null> {
+  try {
+    logMessage('info', 'Fetching Clerk auth token...');
+
+    const authData = auth();
+    const { orgId } = authData;
+    const token = await authData.getToken();
+
+    if (!orgId || !token) {
+      logMessage('error', 'Missing Clerk orgId or token. Authentication failed.');
+      return null;
+    }
+
+    // If orgId exists in cache, return stored GUID (avoid duplicates)
+    if (customerCache.has(orgId)) {
+      logMessage('info', `Customer GUID found in cache for this orgId: ${orgId}`);
+      logCustomerCache(); // Log the cache contents
```

Comment:
Dont log the entire cache content 
Enable debug log to only output org_id and the relevant customer_guid